public/app/api/createBaseQuery.ts

```diff
@@ -0,0 +1,35 @@
+import { lastValueFrom } from 'rxjs';
+
+import { BackendSrvRequest, getBackendSrv, isFetchError } from '@grafana/runtime';
+
+interface RequestOptions extends BackendSrvRequest {
+  manageError?: (err: unknown) => { error: unknown };
+  showErrorAlert?: boolean;
+  body?: BackendSrvRequest['data'];
+}
+
+export function createBaseQuery({ baseURL }: { baseURL: string }) {
```

Comment:
let's type the return as `BaseQueryFn<RequestOptions>`

Reply:
This doesn't infer the error type, which stays `unknown`.

Reply:
I guess we could do `BaseQueryFn<RequestOptions, unknown, Error>`.